app/utils/fetcher.ts

```diff
@@ -0,0 +1,18 @@
+interface Headers {
+  [key: string]: string;
+}
+
+export default async function fetcher<T>(
```

Comment:
Too early for a JSON type? Like:

```ts
type JSONPrimitive = string | number | boolean | JSONObject | null | undefined;
type JSONObject = { [key: string]: JSONPrimitive } | JSONObject[];
```

So you'd do: 
```suggestion
export default async function fetcher<T extends JSONObject>(
```